sgt-back-end/server.js

```diff
@@ -0,0 +1,129 @@
+import express from './node_modules/express/index.js';
+
+import pg from 'pg';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res) => {
+  try {
+    const sql = `
+    select *
+    from "grades";`
+    const result = await db.query(sql)
+    const grades = result.rows;
+      res.status(200).json(grades);
+    } catch (err) {
+    console.error('/api/notes error', err);
+    res.status(500).json({ error: 'an unexpected error occurred' });
+  }});
+
+app.post('/api/grades', async (req, res) => {
+  try {
+    const name = req.body.name;
+    const course = req.body.course;
+    const score = +req.body.score;
+    const params = [name, course, score];
+    const sql = `
+      insert into "grades" ("name", "course", "score")
+      values ($1, $2, $3)
+      returning *;
+      `
+    if (score === undefined ) {
+      res.status(400).json({ error: `missing value in score`});
+    } else if ( course === undefined ) {
+      res.status(400).json({ error: `missing value in grade` });
+    } else if ( name === undefined ) {
+      res.status(400).json({ error: `missing value in name` });
+    } else if ( score < 0 || score > 100 ) {
+      res.status(400).json({ error: `score=${score} is not a valid inter between 0-100` });
+    } else {
+      const results = await db.query(sql, params);
+      const grade = results.rows;
+      res.status(201).json(grade);
```

Comment:
This isn't the correct response. You are returning an array here. You need to return a single object.